### DISCUSSION TITLE
"Snap to the nearest beat" (quantising support after timing changes)

### DISCUSSION THREAD

**Discussion by rvnexe** [ppy/osu#29030] [Ideas] [answered]
Hi. It's common that you might set your timing offset a little off, then comeback and fix it, but unfortunately, your already placed notes will not reposition based on that. Then, your notes will be off from music's rhythm.
<details>

<summary>Like this</summary>

![image](https://github.com/user-attachments/assets/582dc24a-c371-4e6e-9621-922164057d31)

</details>

It takes decades to snap every single one again. So an option to snap all of them automatically will be a life saver. 
Select a range of hit objects, and "Snap selection to the nearest beat"

## Beat division
Imagine a music which is using 1/3, 1/5, ... beat division. Then, this option will break the timings.
So, we will ask user about beat division. (Like how stable's convert to stream works)

## Tolerance
Imagine a music, containing some beats which are not based on any kind of beat division.
If this kind of beat isn't based on a beat division, then it should be far away from your desired division. (at least longer than ~2ms) So, we can have a Tolerance option, so "Don't snap beats which are off more than [...]ms. (Like in my screenshot, which all of them are around 1ms)

**Comment by rvnexe** [comment]
or auto snap them if user changes the offset. who cares

**Comment by peppy** [comment, marked as answer]
This is generally what we call quantising, and stable already has this (although it's quite hard to understand there). Definitely need this to return.
